crates/pathfinder/src/state/sync.rs

```diff
@@ -633,6 +632,10 @@ async fn l2_update(
             }
         }
 
+        // Clear pending data just before we commit to the database. This minimizes the
+        // time between the old latest + pending -> new latest transition.
+        tokio::runtime::Handle::current().block_on(pending_data.clear());
```

Comment:
I don't really like this -- but I'm also not sure how to handle it otherwise?

Reply:
I guess you could use channels to a tokio task that would do the clear and signal back but I doubt it's wort the complexity as the channels in use would have to blocking_send and blocking_recv in the closure that manages the db ops which essentially means calling block_on underneath anyway.

Reply:
Yeah I tried channels at first, but its really clunky. Annoying thing is that sqlite transactions cannot be sent across threads, so always have to open and close them in the same context.